quill-core/src/main/scala/io/getquill/quotation/Parsing.scala

```diff
@@ -353,11 +353,15 @@ trait Parsing extends SchemaConfigParsing {
   private def is[T](tree: Tree)(implicit t: TypeTag[T]) =
     tree.tpe <:< t.tpe
 
+  private def isTraversable(tree: Tree) = {
+    tree.tpe <:< typeOf[Traversable[_]]
+  }
+
   val valueParser: Parser[Value] = Parser[Value] {
     case q"null" => NullValue
     case Literal(c.universe.Constant(v)) => Constant(v)
     case q"((..$v))" if (v.size > 1) => Tuple(v.map(astParser(_)))
-    case q"$pack.Set.apply[..$t](..$v)" => Set(v.map(astParser(_)))
+    case tree @ q"$pack.$coll.apply[..$t](..$v)" if isTraversable(tree) => Set(v.map(astParser(_)))
```

Comment:
Could you rename the ast type `Set` to `Collection`? I think it'll be more consistent with this change